crates/ext-processor/src/service.rs

```diff
@@ -328,6 +395,45 @@
         Ok(plugin_instances)
     }
 
+    async fn execute_init_phase(
+        plugin_instances: Vec<Arc<Mutex<PluginInstance>>>,
+        timeout_duration: std::time::Duration,
+    ) {
+        let mut init_phase_tasks = JoinSet::new();
+        for plugin_instance in plugin_instances.clone() {
```

Comment:
Interestingly, `clippy` was able to spot redundant cloning for `.iter().cloned()` but not `.clone()`? Several of these loops didn't need cloning because they were already cloned elsewhere. Removing redundant clones.